sdk/resource/resource.go

```diff
@@ -21,11 +21,20 @@ import (
 // Resources should be passed and stored as pointers
 // (`*resource.Resource`).  The `nil` value is equivalent to an empty
 // Resource.
+//
+// While Resource is comparable, its comparison should not be relied on when
+// checking equality. The [Resource.Equal] method should be used to check
+// equality between two resources and the [attribute.Distinct] returned from
+// [Resource.Equivalent] should be used for map keys instead. No guantee of the
```

Comment:
Can we actually add this?  Wouldn't it be breaking if resource comparisons stoped working?

Reply:
I'm not intending to say that comparison is going away, just that the result of that comparison is not guaranteed to be stable. Is that not what you are interpreting?

Reply:
Line 36 is being added to ensure the resource will always remain comparable FWIW.

Reply:
I think this is acceptable in Go ecosystem. Especially given that the type already has `Equal` method. 

I would suggest also giving a similar comment to `Equal`. Example: https://pkg.go.dev/time#Time.Equal I mean adding something similar to

> See the documentation on the Time type for the pitfalls of using == with Time values; most code should use Equal instead.

Reply:
> I would suggest also giving a similar comment to `Equal`. Example: https://pkg.go.dev/time#Time.Equal I mean adding something similar to
> 
> > See the documentation on the Time type for the pitfalls of using == with Time values; most code should use Equal instead.

Updated, PTAL.

Reply:
LGTM. I am not resolving so that @dashpole can see the conversation.

Reply:
lgtm